pytorch_lightning/trainer/training_loop.py

```diff
@@ -140,7 +140,7 @@ def on_train_end(self):
         # todo: TPU 8 cores hangs in flush with TensorBoard. Might do for all loggers.
         # It might be related to xla tensors blocked when moving the cpu
         # kill loggers
-        if self.trainer.logger is not None and self.trainer.training_type_plugin.should_finalize:
```

Comment:
I remember that this was added for a reason, mind check the PR which added it?

Reply:
I added this in my previous TPU PR. This PR resolve the bug `should_finalize` prevented.